src/SingleSignatureDirtyHack.sol

```diff
@@ -269,4 +314,90 @@ abstract contract SingleSignatureDirtyHack is IERC5267, AbstractContext {
         transferParams.token.safeTransferFrom(transferParams.from, transferParams.to, transferParams.requestedAmount);
         return true;
     }
+
+    function transferFromPermit(
+        string calldata typeSuffix,
+        TransferParams calldata transferParams,
+        PackedSignature calldata sig
+    ) external returns (bool) {
+        _checkDeadline(transferParams.deadline);
+
+        bytes32 structHash = _hashStruct(typeSuffix, transferParams);
+        bytes32 signingHash = _encodePermitHash(
+            transferParams.token, transferParams.from, transferParams.nonce, transferParams.amount, structHash
+        );
+        
+        address signer;
+        assembly ("memory-safe") {
+            let ptr := mload(0x40)
+            let vs := calldataload(add(0x20, sig))
+            let owner := calldataload(add(0x60, transferParams)) // from
+            let value := calldataload(add(0xa0, transferParams)) // amount
+
+            // get signer
+            mstore(add(0xd4, ptr), and(vs, 0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff)) // s
+            mstore(add(0xb4, ptr), calldataload(sig)) // r
+            mstore(add(0x94, ptr), add(0x1b, shr(0xff, vs))) // v
+            mstore(add(0x74, ptr), signingHash)
+            
+            pop(staticcall(gas(), 0x01, add(0x74, ptr), 0x80, 0x00, 0x20))
+            signer := mul(mload(0x00), eq(returndatasize(), 0x20))
+
+            // keep adding parameters to call permit
+            // TODO: check that calling ecrecover doesn't affect the previously added parameters
+            mstore(add(0x54, ptr), value)
+            mstore(add(0x34, ptr), address())
+            mstore(add(0x14, ptr), owner)
+            mstore(ptr, 0xd505accf000000000000000000000000) // selector for `permit(address,address,uint256,uint256,uint8,bytes32,bytes32)` with `owner` padding
+
+            let token := and(0xffffffffffffffffffffffffffffffffffffffff, calldataload(add(0x40, transferParams)))
```

Comment:
don't clean dirty bits here. dirty bits in calldata should result in a revert with an empty reason. of course, if this checking is performed elsewhere in this context (i.e. the expression `transferParams.token` appears below), then you don't need to clean or revert; we can be confident that the bits are clean

Reply:
dirty bits are killing me haha, why from had dirty bits then in one of your previous comments? it was passed to the function from the calldata

Reply:
ahh it doesn't apply. Got it